File: TryFinally.java
// Enter and Leave a Try Statement with a Finally Handler

public class TryFinally {
	private static boolean shouldThrow = false;

	private static long loop(int times) {
		long startTime = System.nanoTime();
		for (int count = 0 ; count < times ; ++count) {
			try {
				// ...
			} finally {
				// ...
			}
		}
		long endTime = System.nanoTime();
		return endTime - startTime;
	}

	public static void main(String[] args) {
		int times = 1;
		if (0 < args.length) {
			times = Integer.parseInt(args[0]);
		}

		// Warm-Up:
		loop(1000);

		long time = loop(times);
		System.out.format("Run-Time (s): %.1f%n", time / 1_000_000_000.);
	}
}

File: try-finally.py
#!/usr/bin/env python3

# Cross a Try Statement With Finally Clause

from time import thread_time_ns


def main(argv):
    times = 1;
    total_frames = 1;
    if 1 < len(argv):
        times = int(argv[1])
    if 2 < len(argv):
        total_frames = int(argv[2])

    start_time = thread_time_ns()
    for count in range(times):
        try:
            pass
        finally:
            pass

    end_time = thread_time_ns()
    print('Run-Time (s): {:.1f}'.format((end_time - start_time) / 1_000_000_000.))


if '__main__' == __name__:
    import sys
    main(sys.argv)
